src/python/pants/engine/fs.py

```diff
@@ -185,6 +185,7 @@ def create_fs_rules():
   return [
     RootRule(Digest),
     RootRule(DirectoriesToMerge),
+    RootRule(FilesContent),
```

Comment:
If this is exposed for the purposes of testing, rather than for use in the production code, consider declaring the Root rule in the test(s) that consume it instead.

Declaring only the necessary RootRules allows less monomorphization and graph construction in general to happen at runtime.

Reply:
Done